### PR TITLE
[fix](multi-catalog)resolve hive meta store compatibility for different version issues

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

issue: #31442

Fix hive list partition at different version.
Only Hive3 uses the prependCatalogToDbName() to wrap db_name.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run external

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java [hunk 1]
```diff
@@ -931,24 +946,30 @@ public Partition appendPartition(String dbName, String tableName, String partNam
...
     return deepCopy(p);
   }
 
   @Deprecated
   public Partition appendPartition(String dbName, String tableName, List<String> partVals,
                                    EnvironmentContext ec) throws TException {
-    return client.append_partition_with_environment_context(prependCatalogToDbName(dbName, conf),
-        tableName, partVals, ec).deepCopy();
+    if (hiveVersion == HiveVersion.V3_0) {
+      dbName = prependCatalogToDbName(dbName, conf);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java:970]: Unify to `prependCatalogToDbNameByVersion`

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run feut

**Comment by wsjz** [comment]
run fe ut

**Review by morningman — APPROVED** [review]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**